### PR TITLE
Properly fetch shares for files with spaces/other special characters in filenames

### PR DESCRIPTION
**Pull request by claucambra**
Ensure percent encoding is used. PR also includes some other minor fixes

### CODE REVIEW

File: src/gui/filedetails/sharemodel.cpp [hunk 1]
```diff
@@ -253,7 +253,9 @@ void ShareModel::updateData()
     qCDebug(lcShareModel) << "Updating share model data now.";
 
     const auto relPath = _localPath.mid(_folder->cleanPath().length() + 1);
-    _sharePath = _folder->remotePathTrailingSlash() + relPath;
+    const auto sharePath = _folder->remotePathTrailingSlash() + relPath;
+    const auto percentEncoded = QUrl::toPercentEncoding(sharePath);
+    _sharePath = QString::fromUtf8(percentEncoded);
```
Comment (mgallien) [inline src/gui/filedetails/sharemodel.cpp:258]: I would probably move this into a private utility method such that it is easier to grasp what is going on

**Review by mgallien — CHANGES_REQUESTED** [review]

File: src/gui/filedetails/sharemodel.cpp (hunk 1 above)
Comment (claucambra) [reply src/gui/filedetails/sharemodel.cpp]: Done

File: src/gui/sharemanager.cpp [hunk 2]
```diff
@@ -508,29 +509,24 @@ void ShareManager::fetchShares(const QString &path)
...
-        SharePtr newShare;
+    for (const auto &share : tmpShares) {
+        const auto data = share.toObject();
+        const auto shareType = data.value("share_type").toInt();
 
         if (shareType == Share::TypeLink) {
-            newShare = parseLinkShare(data);
-        } else if (Share::isShareTypeUserGroupEmailRoomOrRemote(static_cast <Share::ShareType>(shareType))) {
-            newShare = parseUserGroupShare(data);
+            shares.append(parseLinkShare(data));
```
Comment (allexzander) [inline src/gui/sharemanager.cpp:524]: @claucambra I don't see a good reason for this refactoring, besides removing space after `static_cast` you are basically doing the same but introducing more code for review

**Review by allexzander — CHANGES_REQUESTED** [review]
@claucambra besides one small comment, I've tested and it seems there is some issue with the path:
image

File: src/gui/sharemanager.cpp (hunk 2 above)
Comment (allexzander) [reply src/gui/sharemanager.cpp:524]: while fixing the issue @claucambra would it be possible to write a test for the new method you're adding?

Comment (claucambra) [reply src/gui/sharemanager.cpp:524]: The main reason for this change was to eliminate the likelihood of appending a null `newShare` to `shares`, and making the code a little shorter -- it doesn't seem like we need the `newShare` here. 

As for the space I think this was auto-done by clang-format but I can re-add

Comment (claucambra) [reply src/gui/sharemanager.cpp:524]: Looking into the other points now

Comment (allexzander) [reply src/gui/sharemanager.cpp:524]: @claucambra No, I am fine with removing the space. I can see your point about `newShare` being not `null`, but there is an else statement so it should never be null, that said, you can just add a check `if (newShare)` prior to adding it if that makes sense

**Pull request closed without merging**